packages/expo-updates/android/src/androidTest/java/expo/modules/updates/logging/UpdatesLoggingTest.kt

```diff
@@ -1,49 +1,78 @@
 package expo.modules.updates.logging
 
+import android.os.Bundle
+import androidx.test.platform.app.InstrumentationRegistry
+import expo.modules.core.Promise
+import expo.modules.core.logging.LogType
+import expo.modules.core.logging.PersistentFileLog
+import expo.modules.updates.UpdatesModule
+import expo.modules.updates.logging.UpdatesLogger.Companion.EXPO_UPDATES_LOGGING_TAG
 import expo.modules.updates.logging.UpdatesLogger.Companion.MAX_FRAMES_IN_STACKTRACE
 import junit.framework.TestCase
 import org.junit.Assert
+import org.junit.Before
 import org.junit.Test
 import java.util.*
 
 class UpdatesLoggingTest : TestCase() {
 
+  @Before
+  fun setup() {
+    val instrumentationContext = InstrumentationRegistry.getInstrumentation().context
+    val persistentLog = PersistentFileLog(EXPO_UPDATES_LOGGING_TAG, instrumentationContext)
+    asyncMethodRunning = true
```

Comment:
we'll have to be careful not to ever switch JUnit to run in parallel since that will cause this shared state to be corrupted.

It may be better to have some sort of construct that isolates an instance of the async method waiter to each test. Probably easiest to do this by putting the asyncMethodRunning and waitForAsyncMethodToFinish in a class and instantiate a new instance of that class per-test.

Reply:
Same for the other tests that use this pattern.

Reply:
I'll do that, and I expect we would have to have a different file name for each test

Reply:
Fixed